src/transformers/models/fastspeech2/configuration_fastspeech2.py

```diff
@@ -0,0 +1,154 @@
+
+
+    Args:
+        vocab_size (`int`, *optional*, defaults to 75):
+            Vocabulary size of the FastSpeech2 model. Defines the number of different tokens that can be represented by
+            the `inputs_ids` passed when calling [`~FastSpeech2Model`] or [`~TFFastSpeech2Model`].
+        encoder_layers (`int`, *optional*, defaults to 4):
+            Number of hidden layers in the FastSpeech2 encoder.
+        decoder_layers (`int`, *optional*, defaults to 4):
+            Number of hidden layers in the FastSpeech2 decoder.
+        encoder_attention_heads (`int`, *optional*, defaults to 2):
+            Number of attention heads for each attention layer in the FastSpeech2 encoder.
+        decoder_attention_heads (`int`, *optional*, defaults to 2):
+            Number of attention heads for each attention layer in the FastSpeech2 decoder.
+        intermediate_size (`int`, *optional*, defaults to 3072):
+            Dimension of the "intermediate" (i.e., feed-forward) layer in the Transformer encoder.
+        hidden_dropout_prob (`float`, *optional*, defaults to 0.1):
+            The dropout probabilitiy for all fully connected layers in the embeddings, encoder, and pooler.
+        attention_probs_dropout_prob (`float`, *optional*, defaults to 0.1):
+            The dropout ratio for the attention probabilities.
+        max_source_positions (`int`, *optional*, defaults to 1024):
+            The maximum sequence length that this model might ever be used with. Typically set this to something large
+            just in case (e.g., 512 or 1024 or 2048).
+        initializer_range (`float`, *optional*, defaults to 0.02):
+            The standard deviation of the truncated_normal_initializer for initializing all weight matrices.
+        Example:
+
+    ```python
+    >>> from transformers import FastSpeech2Model, FastSpeech2Config
+
+    >>> # Initializing a FastSpeech2 fastspeech2 style configuration
+    >>> configuration = FastSpeech2Config()
+
+    >>> # Initializing a model from the fastspeech2 style configuration
+    >>> model = FastSpeech2Model(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```"""
+    model_type = "fastspeech2"
+
+    def __init__(
+        self,
+        n_frames_per_step=1,
+        output_frame_dim=80,
+        encoder_embed_dim=256,
+        speaker_embed_dim=64,
+        dropout=0.2,
+        max_source_positions=1024,
+        encoder_attention_heads=2,
+        fft_hidden_dim=1024,
+        fft_kernel_size=9,
+        attention_dropout=0,
+        encoder_layers=4,
+        decoder_embed_dim=256,
+        decoder_attention_heads=2,
+        decoder_layers=4,
+        add_postnet=False,
+        postnet_conv_dim=512,
+        postnet_conv_kernel_size=5,
+        postnet_layers=5,
+        postnet_dropout=0.5,
+        vocab_size=75,
+        num_speakers=1,
+        var_pred_n_bins=256,
```

Comment:
that's quite a lot of variables - We've made the mistake a bit with Wav2Vec2 that we left every parameter modifiable which led to configs becoming huge and a bit unreadable IMO e.g.:  https://huggingface.co/itaihay/wav2vec_asr_swbd/blob/main/config.json

Could we maybe in a first step, hardcode lots of those variables if they don't differ from checkpoint to checkpoint? This will allow us to have shorter, more concise configs. We can always go back and make them modifiable if needed . However we can't do it the other way ;-)

Reply:
Parameters that we don't users to modify we could define as:

```
@property 
def var_pred_n_bins():
    return 256

Reply:
That's a good point. I'll hard code them as `@property`s!

Reply:
Hello @patrickvonplaten, I'm having a bit of trouble deciding on which parameters we want to allow users to modify. Currently, I have `var_pred_num_bins` and `mel_dim`. Do you think we should hard code other parameters too, such as `num_layers` or `kernel_size`?